packages/stores/src/queries/pools/pools.ts

```diff
@@ -98,4 +99,13 @@ export class ObservableQueryPools
       return this.getPool(raw.id)!;
     });
   });
+
+  /** TODO: implement pagination when we hit the limit of pools, for now, the url will be set to the max number of pools in the autorun above */
+  paginate() {
+    /** do nothing */
+  }
+
+  fetchRemainingPools() {
+    /** do nothing since all pools get fetched. */
+  }
```

Comment:
We need to handle this edge case when we have the redesigns pools page @sehyunc. cc\ @jonator

Reply:
Actually I'm thinking the paginate method should be removed across the board since it's not actually used. I added it prematurely to be used in the future, but the way the filtered query works is it fetches a subset of pools, then all of the pools. Pagination is not used or tested. I can remove it.

Reply:
If that's the case let's remove it

Reply:
https://github.com/osmosis-labs/osmosis-frontend/blob/8547c2e0ae5c72d563f13f16095f35c13a48f8e0/packages/web/components/complex/all-pools-table-set.tsx#L525

Oh wait I checked he did use it. I'll keep it around then. For the fallback store, paginate is a noop since all pools should be loaded immediately.

Reply:
The latest version of the branch doesn't have pagination implemented, and performance seems to be very good because of virtualization. I think we could use the query the way @jonator described it and remove pagination for less complexity.